### PR TITLE
audio_core: Resolve compilation warnings

### PR DESCRIPTION
**Pull request by lioncash**
Resolves a few trivial compilation warnings related to truncation, shadowing, and initialization.

### CODE REVIEW

File: src/audio_core/cubeb_sink.cpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ class CubebSinkStream final : public SinkStream {
     u32 num_channels{};
 
     Common::RingBuffer<s16, 0x10000> queue;
-    std::array<s16, 2> last_frame;
+    std::array<s16, 2> last_frame{};
```
Comment (bunnei) [inline src/audio_core/cubeb_sink.cpp]: What's the reason for this change? Isn't std::array already initialized to 0's?

Comment (lioncash) [reply src/audio_core/cubeb_sink.cpp]: std::array by default doesn't initialize primitives. Since `std::array` is defined as an aggregate type, it doesn't have a constructor. The only way elements of a std::array will be initialized is if they have their own constructor that performs some form of initialization. Aside from that, it's possible to request zero-initialization with `{}`.

Comment (bunnei) [reply src/audio_core/cubeb_sink.cpp]: Got it - thanks for the explanation.
